source/features/table-input.tsx

```diff
@@ -0,0 +1,79 @@
+import './table-input.css';
+import React from 'dom-chef';
+import select from 'select-dom';
+import delegate from 'delegate-it';
+import TableIcon from 'octicon/diff-added.svg';
+import * as pageDetect from 'github-url-detection';
+import * as textFieldEdit from 'text-field-edit';
+
+import features from '.';
+
+function generateGfmTable([w, h]: [number, number]): string {
+	return `${'|     '.repeat(w)}|\n${'|-----'.repeat(w)}|\n${`${'|     '.repeat(w)}|\n`.repeat(h)}`.slice(0, -1);
+}
+
+function addTable(event: delegate.Event<MouseEvent, HTMLButtonElement>): void {
+	event.preventDefault();
+	(event.delegateTarget.parentElement!.parentElement as HTMLDetailsElement)!.open = false;
+
+	const field = event.delegateTarget.form!.querySelector('textarea')!;
+	const endsWithNl = field.value.length && !field.value.endsWith('\n');
+	const cursorPos = field.value.length + (endsWithNl ? 4 : 3);
+
+	textFieldEdit.insert(field, (endsWithNl ? '\n' : '') + generateGfmTable(getSquarePos(event.delegateTarget)));
+
+	field.focus();
+	field.setSelectionRange(cursorPos, cursorPos);
+}
+
+function highlightSquares(event: delegate.Event<MouseEvent, HTMLButtonElement>): void {
+	const [x, y] = getSquarePos(event.delegateTarget);
+	[...event.delegateTarget.parentElement!.children].forEach(child => {
+		const [cx, cy] = getSquarePos(child as HTMLElement);
+		child.classList.toggle('rgh-add-table-popup__square--selected', cx <= x && cy <= y);
+	});
+}
+
+function getSquarePos(button: HTMLElement): [number, number] {
+	return button.dataset.position!.split(',', 2).map(Number) as [number, number];
+}
+
+function init(): void {
+	delegate(document, '.rgh-add-table-popup__square', 'click', addTable);
+	delegate(document, '.rgh-add-table-popup__square', 'mouseover', highlightSquares);
```

Comment:
```suggestion
	delegate(document, '.rgh-add-table-popup__square', 'mouseenter', highlightSquares);
```

Reply:
I think I tried `mouseenter` first, but for some reason it didn't work.

Reply:
> `mouseover` is fired every time the mouse moves

Actually that's [`mousemove`](https://developer.mozilla.org/en-US/docs/Web/API/Element/mousemove_event). Tested again with `mouseenter`, doesn't work (because of the embedded `<div>` "covering" each button), so I'll keep `mouseover`.

Reply:
Don't "resolve" suggestions when they aren't resolved because then I have to go find it again and re-open it.

`mouseenter` works, you just need to use `{capture: true}` as an additional parameter: https://stackoverflow.com/questions/50177348/mouseenter-delegation-using-vanilla-javascript

```js
	delegate(document, '.rgh-add-table-popup__square', 'mouseenter', highlightSquares, {capture: true});
```

Reply:
Sorry about that. But what does it matter if we use `mouseover` instead of `mouseenter` here? (not being dismissive, this is a real question)

Reply:
Applied the change to move things along, but I'm still curious why `mouseenter` was a better choice here than `mouseover`.

Reply:
Mousemove is fired every time the mouse moves. Delegate still has to filter all those events through the selector. 

Mouseenter is the one that fires when you enter an element, so *it’s the right tool for the job* in the first place 😁